File: LeetCode/lonelyNum.java
import java.util.*;

class lonelyNum {
    public List<Integer> findLonely(int[] nums) {
        Map<Integer, Integer> obj = new HashMap<>();
        List<Integer> arr = new ArrayList<>();

        // Populate the frequency map
        for (int num : nums) {
            obj.put(num, obj.getOrDefault(num, 0) + 1);
        }

        // Check for lonely numbers
        for (int num : nums) {
            if (obj.get(num) == 1 && !obj.containsKey(num - 1) && !obj.containsKey(num + 1)) {
                arr.add(num);
            }
        }

        return arr;
    }

    public static void main(String[] args) {
        lonelyNum sol = new lonelyNum();
        int[] nums = {10, 6, 5, 8};
        List<Integer> result = sol.findLonely(nums);
        System.out.println(result); // Output: [10, 8]
    }
}

File: LeetCode/lonelyNum.js
function findLonely(nums) {
    const freq = {};
    const result = [];

    // Populate the frequency map
    for (const num of nums) {
        if (freq[num] === undefined) {
            freq[num] = 0;
        }
        freq[num]++;
    }

    for (const num of nums) {
        if (
            freq[num] === 1 && 
            (freq[num - 1] === undefined ) &&
            (freq[num + 1] === undefined )
        ) {
            result.push(num);
        }
    }

    return result;
}

// Example 1
console.log(findLonely([10, 6, 5, 8])); // Output: [10, 8]

// Example 2
console.log(findLonely([1, 3, 5, 3])); // Output: [1, 5]


//approach 2


var findLonely = function(nums) {
    nums.sort((a, b) => a - b);  // Sort the array
    let list = [];  // Initialize an empty list

    for (let i = 0; i < nums.length; i++) {
        let isLonely = true;

        // Check the left neighbor
        if (i > 0 && (nums[i] === nums[i - 1] + 1 || nums[i] === nums[i - 1])) {
            isLonely = false;
        }

        // Check the right neighbor
        if (i < nums.length - 1 && (nums[i] === nums[i + 1] - 1 || nums[i] === nums[i + 1])) {
            isLonely = false;
        }

        // If the number is lonely, add it to the list
        if (isLonely) {
            list.push(nums[i]);
        }
    }

    return list;
};

// Example usage:
console.log(findLonely([10, 6, 5, 8])); // Output: [10, 8]
